onboarding page is done

sefinehtesfa34 opened the pull request:
![image](https://user-images.githubusercontent.com/88246220/233827486-d0a90619-4eaa-4b44-b56c-e753d3ea5b52.png)
Updates after reviewed:
1. Style naming on colors, inside core/shared_widgets/colors.dart file
2. Removed the 2D array of widget sizes.
3. Modularized files

Redi22 reviewed (changes requested):
Here are some general rules you should follow when you create a PR:

- Writing Good PR descriptions
- Don't make changes to other groups file
- Define colors in a single file inside core that way everyone can use the same colors
- Avoid giving sizes like this BackgroundContainer(top: 200, left: 300), use MediaQuery.of(context).size.height this way it will work the same for different screen sizes.

I have seen that you have made changes to the config files which should not be here because they might be a problem for other people after merging. Please revert these changes and push back only files changed that are related to the specific PR.

sefinehtesfa34 commented:
I didn't see any change to the config files. Can you tell me specifically
which file was changed, please?

On Sun, Apr 23, 2023 at 11:40 AM Redi_D ***@***.***> wrote:

> ***@***.**** requested changes on this pull request.
>
> Here are some general rules you should follow when you create a PR:
>
>    - Writing Good PR descriptions
>    - Don't make changes to other groups file
>    - Define colors in a single file inside core that way everyone can use
>    the same colors
>    - Avoid giving sizes like this BackgroundContainer(top: 200, left:
>    300), use MediaQuery.of(context).size.height this way it will work the same
>    for different screen sizes.
>
> I have seen that you have made changes to the config files which should
> not be here because they might be a problem for other people after merging.
> Please revert these changes and push back only files changed that are
> related to the specific PR.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/RealEskalate/Internship-2023/pull/30#pullrequestreview-1396867066>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AVBIPTE4RF6LOETV4XCSSLDXCTTIDANCNFSM6AAAAAAXILXFRM>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

Redi22 commented on Mobile/starter_project_dartsmiths/.dart_tool/package_config.json:
The files in the .dart_tool filder should not be committed.

YiseBoge commented:
Some general points
I like the comments @Redi22 left for you about your git usage
- we are not using the `[team] commit msg` format anymore. please check out the guideline documents or just google `conventional commit` rules
- stop using past tense descriptions in the commit messages as well (you don't need to change these now, you can make that edit when you do squash and merge at the end of this PR)
- make the PR title also fit with the conventional commit rules (because it will be the default commit message used when doing squash and merge)
- please check the files changed part of the PR to see that what you are going to merge is as you want it (when creating a PR)
- minor point but try to stick with kebab-case in your branch naming

YiseBoge commented on Mobile/starter_project_dartsmiths/lib/features/onboarding/onboardingLowerLayout.dart:
```diff
@@ -0,0 +1,43 @@
+import 'package:flutter/material.dart';
+class LandingPageLowerImage extends StatelessWidget {
```
[nit] should we have one line space between imports and the rest?

Redi22 replied:
I would also like the name changed. there is no landing page in the app it is onboarding page and the name can be more formal also.

YiseBoge commented on Mobile/starter_project_dartsmiths/lib/features/onboarding/onboardingUpperLayout.dart:
```diff
@@ -0,0 +1,37 @@
+import 'package:flutter/material.dart';
+class LandingPageUpperImage extends StatelessWidget {
```
[nit] space after imports again

YiseBoge commented on Mobile/starter_project_dartsmiths/pubspec.yaml:
```diff
@@ -59,10 +59,8 @@ flutter:
   uses-material-design: true
 
   # To add assets to your application, add an assets section, like this:
```
[very-nit] it might be better to also get rid of the instruction comment 
or modify it to look like it is describing what is already done

Redi22 replied:
These are files flutter generates so we can fix these to describe our things more but I don't see the point in him changing this or even in getting rid of them.

Redi22 replied:
These are files flutter generates so we can fix these to describe our things more but I don't see the point in him changing this or even in getting rid of them.

YiseBoge reviewed (commented):
I have left some comments but take it with a grain of salt because i am not a flutter expert

Redi22 commented on Mobile/starter_project_dartsmiths/lib/core/shared_widgets/colors.dart:
```diff
@@ -0,0 +1,8 @@
+import 'package:flutter/material.dart';
+const onBoardingBackgroundColor = Color(0XFFE6EAF1);
+const onBoardingArticlePageColor = Color(0XFFFFFFFF);
+const nextPointerColor1 = Color(0XFF376AED);
```
@sefinehtesfa34 check the section "Extended dart" on the flutter guideline for the naming conventions, please.

Redi22 commented on Mobile/starter_project_dartsmiths/lib/features/onboarding/onBoardingPage.dart:
```diff
@@ -0,0 +1,36 @@
+
+class OnboardingPage extends StatelessWidget {
+  const OnboardingPage(
+      {super.key, });
```
no, it is not. It is better to remove it.

BekyH commented on Mobile/starter_project_dartsmiths/lib/core/shared_widgets/colors.dart:
move this file to the utils directory.

BekyH commented on Mobile/starter_project_dartsmiths/lib/features/onboarding/articlePage.dart:
```diff
@@ -0,0 +1,77 @@
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
```
Please take look at the Flutter best practices doc section 9 regarding containers.

Redi22 replied:
@BekyH can you specify the thing that needs fixing first and then suggest the section please? It will be better and faster to fix.

BekyH replied:
Container widget is hefty weight will have an adverse effect on performance. Therefore, using alternate widgets like sizedBox or decoratedBox is recommended.

Redi22 replied:
That makes sense but keep in mind it is a preference for widgets not a must currently.

BekyH commented on Mobile/starter_project_dartsmiths/lib/features/onboarding/nextPointer.dart:
```diff
@@ -0,0 +1,70 @@
+        Row(
+          mainAxisAlignment: MainAxisAlignment.spaceBetween,
+          children: [
+            Container(
```
same as the above containers.

BekyH commented on Mobile/starter_project_dartsmiths/lib/features/onboarding/nextPointer.dart:
```diff
@@ -0,0 +1,70 @@
+                color: widget.colors[index][1],
+              ),
+            ),
+            Container(
```
same as the above containers.

BekyH commented on Mobile/starter_project_dartsmiths/lib/features/onboarding/onBoardingPage.dart:
```diff
@@ -0,0 +1,36 @@
+  Widget build(BuildContext context) {
+    return Stack(
+      children: [
+        Container(
```
same as the above containers.

BekyH commented on Mobile/starter_project_dartsmiths/lib/features/onboarding/onboardingLowerLayout.dart:
```diff
@@ -0,0 +1,43 @@
+        children: [
+          Padding(
+            padding: const EdgeInsets.only(right: 10),
+            child: Container(
```
same as the above containers.

BekyH commented on Mobile/starter_project_dartsmiths/lib/features/onboarding/onboardingLowerLayout.dart:
```diff
@@ -0,0 +1,43 @@
+            ),
+          ),
+          Expanded(
+            child: Container(
```
same as the above containers.

BekyH commented on Mobile/starter_project_dartsmiths/lib/features/onboarding/onboardingUpperLayout.dart:
```diff
@@ -0,0 +1,37 @@
+      children: [
+        Padding(
+          padding: const EdgeInsets.only(right: 10.0),
+          child: Container(
```
same as the above containers.

sefinehtesfa34 replied:
I am using decorations. SizedBox can't help me for such like styles.

BekyH commented on Mobile/starter_project_dartsmiths/lib/features/onboarding/onboardingUpperLayout.dart:
```diff
@@ -0,0 +1,37 @@
+          ),
+        ),
+        Expanded(
+          child: Container(
```
same as the above containers.

YiseBoge commented on Mobile/starter_project_dartsmiths/.gitignore:
```diff
@@ -42,3 +42,6 @@ app.*.map.json
 /android/app/release
+
+.fake
+.dart_tool/package_config.json
```
and since you have already commited changes in that folder (and .vscode folder too, and android folder too), you will have to add another commit to revese those changes before adding them to git ignore

Redi22 replied:
I don't think the android folder should be ignored. Later on there will be build changes and config changes to those files, I agree on ignoring the .dart_tools and .vs_code folders though.

YiseBoge commented on Mobile/starter_project_dartsmiths/.dart_tool/package_config.json:
i think this should be untracked still

Redi22 replied:
it should

lealemgetahun commented on Mobile/starter_project_dartsmiths/assets/images/image1.jpg:
use meaning full names for the images

lealemgetahun commented on Mobile/starter_project_dartsmiths/lib/core/shared_widgets/colors.dart:
```diff
@@ -0,0 +1,7 @@
+import 'package:flutter/material.dart';
+const onBoardingBackgroundColor = Color(0XFFE6EAF1);
```
Rebase from main and use the colors defined there

lealemgetahun commented on Mobile/starter_project_dartsmiths/lib/features/onboarding/article_content.dart:
```diff
@@ -0,0 +1,57 @@
+            topLeft: Radius.circular(24.0),
+            topRight: Radius.circular(24.0),
+          ),
+          color: Color(0xFFFFFFFF)),
```
Avoid defining colors like this. Use the color constants

lealemgetahun commented on Mobile/starter_project_dartsmiths/lib/features/onboarding/article_content.dart:
```diff
@@ -0,0 +1,57 @@
+          color: Color(0xFFFFFFFF)),
+      child: Padding(
+        padding:
+            const EdgeInsets.only(left: 40, right: 60, bottom: 10.0, top: 10.0),
```
use media query for sizes

sefinehtesfa34 commented:
Updates:
I have changed the hard-coded padding to media query based.
I have changed the name of the image files.
I rebased the main branch and used the color constants.

On Wed, Apr 26, 2023 at 9:34 AM lealemgetahun ***@***.***>
wrote:

> ***@***.**** requested changes on this pull request.
> ------------------------------
>
> On Mobile/starter_project_dartsmiths/assets/images/image1.jpg
> <https://github.com/RealEskalate/Internship-2023/pull/30#discussion_r1177403927>
> :
>
> use meaning full names for the images
> ------------------------------
>
> In Mobile/starter_project_dartsmiths/lib/core/shared_widgets/colors.dart
> <https://github.com/RealEskalate/Internship-2023/pull/30#discussion_r1177404371>
> :
>
> > @@ -0,0 +1,7 @@
> +import 'package:flutter/material.dart';
> +const onBoardingBackgroundColor = Color(0XFFE6EAF1);
>
> Rebase from main and use the colors defined there
> ------------------------------
>
> In
> Mobile/starter_project_dartsmiths/lib/features/onboarding/article_content.dart
> <https://github.com/RealEskalate/Internship-2023/pull/30#discussion_r1177404818>
> :
>
> > +  @override
> +  State<ArticleContent> createState() => _ArticlePageState();
> +}
> +
> +class _ArticlePageState extends State<ArticleContent> {
> +  @override
> +  Widget build(BuildContext context) {
> +    return Container(
> +      height: MediaQuery.of(context).size.height /3,
> +      width: MediaQuery.of(context).size.width,
> +      decoration: const BoxDecoration(
> +          borderRadius: BorderRadius.only(
> +            topLeft: Radius.circular(24.0),
> +            topRight: Radius.circular(24.0),
> +          ),
> +          color: Color(0xFFFFFFFF)),
>
> Avoid defining colors like this. Use the color constants
> ------------------------------
>
> In
> Mobile/starter_project_dartsmiths/lib/features/onboarding/article_content.dart
> <https://github.com/RealEskalate/Internship-2023/pull/30#discussion_r1177405148>
> :
>
> > +
> +class _ArticlePageState extends State<ArticleContent> {
> +  @override
> +  Widget build(BuildContext context) {
> +    return Container(
> +      height: MediaQuery.of(context).size.height /3,
> +      width: MediaQuery.of(context).size.width,
> +      decoration: const BoxDecoration(
> +          borderRadius: BorderRadius.only(
> +            topLeft: Radius.circular(24.0),
> +            topRight: Radius.circular(24.0),
> +          ),
> +          color: Color(0xFFFFFFFF)),
> +      child: Padding(
> +        padding:
> +            const EdgeInsets.only(left: 40, right: 60, bottom: 10.0, top: 10.0),
>
> use media query for sizes
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/RealEskalate/Internship-2023/pull/30#pullrequestreview-1401254494>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AVBIPTHKINTV37QAL6BSIGDXDC6V7ANCNFSM6AAAAAAXILXFRM>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

Redi22 reviewed (approved):
LGTM

BekyH reviewed (dismissed):
LG but also check the folders again. onboarding folder is in features directory and also out side of features directory. It should be inside features directory.
